### PR TITLE
Expansion fixes

### PR DESCRIPTION
**Pull request by brinxmat**
- Note:
  - label -> name (in UnconfirmedPlace and Event)
  - countryCode -> country (not necessary TBH, but "code" is weird and wasn't working)
  - Otherwise, small fixes to make the JSON-LD strictly compliant (a result of changing the algorithm implementation)
  - Improving testing to avoid syntactic errors

### CODE REVIEW

File: expansion/src/test/java/no/unit/nva/publication/indexing/ExpandedResourceTest.java [hunk 1]
```diff
@@ -161,7 +161,7 @@ void shouldReturnIndexDocumentWithValidContributorAffiliationCountryCode() throw
         var indexDocument = fromPublication(fakeUriRetriever, resourceService, publication);
         var framedResultNode = indexDocument.asJsonNode();
 
-        var actualCountryCode = framedResultNode.at("/entityDescription/contributors/1/affiliations/0/countryCode")
+        var actualCountryCode = framedResultNode.at("/entityDescription/contributors/1/affiliations/0/country")
```
Comment (brinxmat) [inline expansion/src/test/java/no/unit/nva/publication/indexing/ExpandedResourceTest.java:164]: We have changed this… @elkebab probably relevant for you.

File: publication-event-handlers/src/main/java/no/unit/nva/publication/events/handlers/expandresources/ExpandDataEntriesHandler.java [hunk 2]
```diff
@@ -124,7 +124,7 @@ private EventReference processDataEntryUpdateEvent(DataEntryUpdateEvent blobObje
     }
 
     private EventReference createEnrichedEventReference(Entity newData) {
-        return attempt(() -> resourceExpansionService.expandEntry(newData))
+        return attempt(() -> resourceExpansionService.expandEntry(newData, true))
```
Comment (brinxmat) [inline publication-event-handlers/src/main/java/no/unit/nva/publication/events/handlers/expandresources/ExpandDataEntriesHandler.java:127]: Could also be false

File: publication-model/src/main/java/no/unit/nva/api/PublicationContext.java [hunk 3]
```diff
@@ -4,10 +4,13 @@
...
 import no.unit.nva.model.Publication;
+import nva.commons.core.Environment;
+import nva.commons.core.paths.UriWrapper;
 
 public class PublicationContext {
 
     public static JsonNode getContext(Publication publication) {
-        return attempt(() -> dataModelObjectMapper.readTree(publication.getJsonLdContext())).orElseThrow();
+        return attempt(() -> dataModelObjectMapper.readTree(
+            Publication.getJsonLdContext(UriWrapper.fromHost(new Environment().readEnv("API_HOST")).getUri()))).orElseThrow();
```
Comment (brinxmat) [inline publication-model/src/main/java/no/unit/nva/api/PublicationContext.java:14]: Right now, the URI does nothing.

File: publication-model/src/main/java/no/unit/nva/model/contexttypes/Event.java [hunk 4]
```diff
@@ -14,41 +15,43 @@
...
-    public static final String PLACE = "place";
-    public static final String TIME = "time";
-    public static final String AGENT = "agent";
-    public static final String PRODUCT = "product";
-    public static final String SUB_EVENT = "subEvent";
-    @JsonProperty(LABEL)
-    private final String label;
-    @JsonProperty(PLACE)
+    private static final String LABEL_FIELD = "label";
+    private static final String NAME_FIELD = "name";
```
Comment (brinxmat) [inline publication-model/src/main/java/no/unit/nva/model/contexttypes/Event.java:19]: We now use "name" not "label" @elkebab

File: publication-model/src/main/java/no/unit/nva/model/contexttypes/place/UnconfirmedPlace.java [hunk 5]
```diff
@@ -1,50 +1,39 @@
...
 import com.fasterxml.jackson.annotation.JsonProperty;
 import com.fasterxml.jackson.annotation.JsonTypeInfo;
 import java.util.Objects;
 import nva.commons.core.JacocoGenerated;
 
 @JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "type")
-public class UnconfirmedPlace implements Place {
-    public static final String COUNTRY = "country";
-    public static final String LABEL = "label";
+public record UnconfirmedPlace(@JsonAlias(LABEL_FIELD) @JsonProperty(NAME_FIELD) String name,
```
Comment (brinxmat) [inline publication-model/src/main/java/no/unit/nva/model/contexttypes/place/UnconfirmedPlace.java:10]: Similarly, we have replaced "label" with "name" @elkebab

File: expansion/src/main/java/no/unit/nva/expansion/ResourceExpansionServiceImpl.java [hunk 6]
```diff
@@ -65,12 +67,17 @@ public ResourceExpansionServiceImpl(ResourceService resourceService,
     }
 
     @Override
-    public ExpandedDataEntry expandEntry(Entity dataEntry) throws JsonProcessingException, NotFoundException {
+    public ExpandedDataEntry expandEntry(Entity dataEntry, boolean replaceContext) throws JsonProcessingException,
```
Comment (monaullah) [inline expansion/src/main/java/no/unit/nva/expansion/ResourceExpansionServiceImpl.java:70]: Rename to `inlineUriContext` or `uriContext`? With `replaceContext`, I was asking the question: "To what?"

File: expansion/src/main/java/no/unit/nva/expansion/ResourceExpansionServiceImpl.java [hunk 7]
```diff
@@ -79,7 +86,7 @@ public ExpandedDataEntry expandEntry(Entity dataEntry) throws JsonProcessingExce
         } else if (dataEntry instanceof Message message) {
             logger.info("Expanding Message: {}", message.getIdentifier());
             var ticket = ticketService.fetchTicketByIdentifier(message.getTicketIdentifier());
-            return expandEntry(ticket);
+            return expandEntry(ticket, true);
```
[the remark this answers is not in the document]
Comment (monaullah) [reply expansion/src/main/java/no/unit/nva/expansion/ResourceExpansionServiceImpl.java:89]: If `true` means using uri context, I vote for `true`

File: expansion/src/main/java/no/unit/nva/expansion/model/ExpandedResource.java [hunk 8]
```diff
@@ -372,16 +373,17 @@ private static ObjectNode transformToJsonLd(Publication publication) throws Json
...
 
     private static URI extractJsonLdId(Publication publication) {
         return UriWrapper.fromUri(PUBLICATION_HOST_URI).addChild(publication.getIdentifier().toString()).getUri();
     }
 
-    private static JsonNode extractJsonLdContext(Publication publication) {
-        var jsonContext = publication.getJsonLdContext();
+    private static JsonNode extractJsonLdContext() {
+        var jsonContext = Publication.getJsonLdContext(
+            UriWrapper.fromHost(new Environment().readEnv("API_HOST")).getUri());
```
Comment (monaullah) [inline expansion/src/main/java/no/unit/nva/expansion/model/ExpandedResource.java:386]: Make "API_HOST" constant?

Comment (monaullah) [reply expansion/src/main/java/no/unit/nva/expansion/model/ExpandedResource.java:386]: Maybe this already is defined as a constant in some common place?

File: expansion/src/main/java/no/unit/nva/expansion/utils/SearchIndexFrame.java [hunk 9]
```diff
@@ -1,23 +1,38 @@
...
-            FRAME_SRC = JsonDocument.of(IoUtils.inputStreamFromResources("frame.json"));
-        } catch (JsonLdError e) {
-            throw new RuntimeException(e);
-        }
+    public static JsonDocument getFrameWithContext(Path framePath) {
+        return attempt(() -> JsonDocument.of(generateFrameWithContext(framePath))).orElseThrow();
+    }
+
+    private static InputStream generateFrameWithContext(Path framePath) {
+        var uri = UriWrapper.fromHost(new Environment().readEnv("API_HOST")).getUri();
```
Comment (monaullah) [inline expansion/src/main/java/no/unit/nva/expansion/utils/SearchIndexFrame.java:24]: Make "API_HOST" constant?

File: expansion/src/test/java/no/unit/nva/publication/indexing/ExpandedResourceTest.java (hunk 1 above)
Comment (monaullah) [reply expansion/src/test/java/no/unit/nva/publication/indexing/ExpandedResourceTest.java:164]: This requires updating a json pointer in `nva-nvi` as well. We should coordinate deployment

File: publication-model/src/main/java/no/unit/nva/api/PublicationContext.java (hunk 3 above)
Comment (monaullah) [inline publication-model/src/main/java/no/unit/nva/api/PublicationContext.java:14]: Make "API_HOST" constant?

**Review by truhacevkir — APPROVED** [review]

File: expansion/src/main/java/no/unit/nva/expansion/model/ExpandedResource.java (hunk 8 above)
Comment (truhacevkir) [reply expansion/src/main/java/no/unit/nva/expansion/model/ExpandedResource.java:386]: Maybe put it to ExpansionConfig file?

File: expansion/src/main/java/no/unit/nva/expansion/ResourceExpansionServiceImpl.java (hunk 6 above)
Comment (truhacevkir) [reply expansion/src/main/java/no/unit/nva/expansion/ResourceExpansionServiceImpl.java:70]: What about shouldReplaceContext?

File: expansion/src/test/java/no/unit/nva/publication/indexing/ExpandedResourceTest.java (hunk 1 above)
Comment (brinxmat) [reply expansion/src/test/java/no/unit/nva/publication/indexing/ExpandedResourceTest.java:164]: Ok, will make it "countryCode"

**Pull request merged**